source/Pacioli/Pacioli.Tests/Tests/AuthorizationAndAuthenticationTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Moq;
+using Pacioli.WebApi.Controllers;
+using Pacioli.WebApi.Models;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Pacioli.Tests.Tests
+{
+    public class AuthorizationAndAuthenticationTests
+    {
+        private readonly Mock<IConfiguration> _configuration;
+
+        public AuthorizationAndAuthenticationTests()
+        {
+            _configuration = new Mock<IConfiguration>();
+            _configuration.SetupGet(c => c["JWT:ValidIssuer"]).Returns("https://localhost:5001");
+            _configuration.SetupGet(c => c["JWT:ValidAudience"]).Returns("https://localhost:5001");
+            _configuration.SetupGet(c => c["JWT:Secret"]).Returns("Very long secret key required by encryption algorithm");
+        }
+
+        [Fact]
+        public async Task AuthorizeAccountantWithCorrectCredentials()
+        {
+            UserController sut = new(MockUserManager<Accountant>(users).Object, _configuration.Object);
+
+            IActionResult result = await sut.Login(new LoginModel { Email = "fakeEmail@fakes.com", Password = "fakePassword" });
+
+            Assert.IsType<OkObjectResult>(result);
+        }
+
+        [Fact]
+        public async Task RejectAccountantWithInCorrectCredentials()
+        {
+            UserController sut = new(MockUserManager<Accountant>(users, false).Object, _configuration.Object);
+
+            IActionResult result = await sut.Login(new LoginModel { Email = "fakeEmail@fakes.com", Password = "fakePassword" });
+
+            Assert.IsType<UnauthorizedResult>(result);
+        }
+
+        public static Mock<UserManager<Accountant>> MockUserManager<TUser>(List<TUser> fakeUserStore, bool isValidUser = true)
```

Comment:
This method is a bit busy, do you think you can split it up into a few shorter well-named methods?